scene/main/node.cpp

```diff
@@ -1077,16 +1065,11 @@ void Node::_generate_serial_child_name(const Node *p_child, StringName &name) co
 
 	for (;;) {
 		StringName attempt = name_string + nums;
+
 		bool exists = false;
 
-		for (int i = 0; i < cc; i++) {
-			if (children_ptr[i] == p_child) {
-				continue;
-			}
-			if (children_ptr[i]->data.name == attempt) {
-				exists = true;
-			}
-		}
+		existing = data.children.getptr(attempt);
+		exists = existing != nullptr && *existing != p_child;
```

Comment:
The declaration above (`bool exists = false;`) could be removed:
```suggestion
		bool exists = existing != nullptr && *existing != p_child;
```